op-acceptance-tests/tests/interop/loadtest/interop_load_test.go

```diff
@@ -1,170 +1,175 @@
+// Package loadtest contains interop load tests.
+//
+// Set NAT_INTEROP_LOADTEST_TARGET to the initial amount of messages that should be passed per L2 slot in each test (default: 100).
+//
+// Each test increases the message throughput until some threshold is reached (e.g., the gas target).
+// The throughput is decreased if the threshold is exceeded or if errors are encountered (e.g., transaction inclusion failures).
 package loadtest
 
 import (
-	"math"
 	"math/rand"
 	"os"
+	"path/filepath"
 	"strconv"
 	"sync"
 	"testing"
 	"time"
 
-	"github.com/ethereum-optimism/optimism/devnet-sdk/contracts/constants"
-	"github.com/ethereum-optimism/optimism/op-acceptance-tests/tests/interop"
 	"github.com/ethereum-optimism/optimism/op-devstack/devtest"
 	"github.com/ethereum-optimism/optimism/op-devstack/dsl"
 	"github.com/ethereum-optimism/optimism/op-devstack/presets"
 	"github.com/ethereum-optimism/optimism/op-service/eth"
-	"github.com/ethereum-optimism/optimism/op-service/plan"
-	"github.com/ethereum-optimism/optimism/op-service/txintent"
-	"github.com/ethereum-optimism/optimism/op-service/txplan"
-	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
-	"github.com/ethereum/go-ethereum/common"
 	"golang.org/x/net/context"
 )
 
 func TestMain(m *testing.M) {
 	presets.DoMain(m, presets.WithSimpleInterop())
 }
 
-// TestLoad passes messages from one chain to another.
-// It assumes that both chains have the same block time.
-// Set NAT_INTEROP_LOADTEST_TARGET to the initial amount of messages that should be passed per block time.
-// The test will run until the test deadline.
-func TestLoad(gt *testing.T) {
+// TestSteady attempts to slightly exceed the gas target in every block by spamming interop messages,
+// simulating benign but heavy activity.
+// Set NAT_INTEROP_LOADTEST_STEADY_TIMEOUT to configure TestSteady's timeout.
+// If unset, the test will run until the go test timeout before exiting successfully. Also see: https://github.com/golang/go/issues/48157.
+func TestSteady(gt *testing.T) {
+	t := setupT(gt)
+	var wg sync.WaitGroup
+	defer wg.Wait()
+	ctx := t.Ctx()
+	if timeoutStr, exists := os.LookupEnv("NAT_INTEROP_LOADTEST_STEADY_TIMEOUT"); exists {
+		timeout, err := time.ParseDuration(timeoutStr)
+		t.Require().NoError(err)
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		t.Cleanup(cancel) // We only care about the timeout.
+	}
+	aimd, source, dest := setupLoadTest(t, ctx, &wg)
+	elasticityMultiplier := dest.Config.ElasticityMultiplier()
+	for range aimd.Ready() {
+		wg.Add(1)
+		go func() {
```

Comment:
Behavior change alert: worker pool is now unbounded. In my local tests, the size of the worker pool was often a bottleneck, so I decided to remove it as part of the refactor.